env0/resource_aws_credentials.go

```diff
@@ -2,12 +2,36 @@ package env0
 
 import (
 	"context"
+	"regexp"
 
 	"github.com/env0/terraform-provider-env0/client"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+var externalIdRegex *regexp.Regexp = regexp.MustCompile(`^[A-Za-z0-9+=\-.,@:/]{2,1000}$`)
+
+func IsExternalIdValid(d *schema.ResourceData, apiClient client.ApiClientInterface) (bool, error) {
+	isSelfHosted, err := apiClient.IsOrganizationSelfHostedAgent()
+
+	if err != nil {
+		return false, err
+	}
+
+	if isSelfHosted {
+		// Validating external id isn't required in self hosted organizations.
+		return true, nil
+	}
+
+	externalId, ok := d.GetOk("external_id")
+	if !ok || externalId.(string) == "" {
```

Comment:
why is external-id empty is valid in case of the schema do contain this property?